### ISSUE TITLE
Parser does not handle OPML 2.0 categories

### ISSUE THREAD

**Issue by neroux** [rometools/rome#462]
Remotely related to #262.

While 2.0 categories are supported during writing since that ticket, it appears parsing still does not support them. That seems to be because the parsing method mostly delegates this to its 1.0 counterpart, where categories are not supported at all.

https://github.com/rometools/rome/blob/ac7d9c83b5123096e5d75260f2b1143c2134f9ed/rome-opml/src/main/java/com/rometools/opml/io/impl/OPML20Parser.java#L72-L90

**Comment by neroux** [comment]
Further looking into the issue it would seem as if the `parseOutline` method would just need to handle the `category` attribute.

A quick fix might be

```java
    protected Outline parseOutline(final Element e, final boolean validate, final Locale locale) throws FeedException {
        Outline retValue;

        retValue = super.parseOutline(e, validate, locale);

        if (e.getAttributeValue("created") != null) {
            retValue.setCreated(DateParser.parseRFC822(e.getAttributeValue("created"), locale));
        }

        final List<Attribute> atts = retValue.getAttributes();

        for (int i = 0; i < atts.size(); i++) {
            final Attribute a = atts.get(i);

            if (a.getName().equals("created")) {
                retValue.getAttributes().remove(a);

                break;
            }

            if (a.getName().equals("category"))
            {
                String categories = a.getValue();
                if (categories != null) retValue.setCategories(Arrays.asList(categories.split(",")));
                break;
            }
        }

        return retValue;
    }
```

However considering there are quite a few contributions pending I am not sure if there is a point in officially providing that fix. @farrukhnajmi @mishako @PatrickGotthard @snoopdave, could you please comment on the project's status?

**Comment by snoopdave** [comment]
I'm still a ROME user and contributor but I haven't made a change in years
and I don't have the keys necessary to publish releases to Maven Central. I
should have some time to review the pull requests over the next couple of
weeks.

Dave

**Comment by PatrickGotthard** [comment]
Hi dave,

there's a GitHub action that automatically releases to Maven Central. You should have the permissions to execute it ;)

Regards,
Patrick

**Comment by neroux** [comment]
@PatrickGotthard, would that mean you have sort of put the project on hold for now?

**Comment by snoopdave** [comment]
Patrick can respond to this, but IMHO:

It means that there are at least a couple of people who can review PRs and
make a ROME release happen if you/we need one. But, you are right that the
current developers have mostly moved on to other things and are not
actively doing development.

Dave

**Comment by PatrickGotthard** [comment]
Hi @neroux,

we already mention on the project page that this project is currently not actively maintained.

With a demanding job and two children I simply do not have enough time.

Also I never really had an in-depth knowledge about RSS and Atom. I became the project lead because the project was completely unmaintained and it was stuck on a really old Java version that didn't support generics as far as I remember ;)

Regards, 
Patrick

**Comment by snoopdave** [comment]
Great! Thanks for setting that up. I will probably kick out a release once
I get some of the PRs merged. Are there any does for how to make a release?

Dave

**Comment by PatrickGotthard** [comment]
Hi @snoopdave,

to create a new release:

- checkout the latest master
- execute release.cmd or release.sh
- push the changes to GitHub

The scripts are self-explaining, just execute them :)

Regards,
Patrick

**Comment by neroux** [comment]
Incorporated aforementioned patch in #465

**Comment by neroux** [comment]
Incorporated aforementioned patch in #466

**Comment by neroux** [comment]
@snoopdave @PatrickGotthard Any thoughts on #466?

**Comment by PatrickGotthard** [comment]
Hi @snoopdave / @mishako,

releasing a new version is now much easier. Open the **Actions** tab > Select **Release Build** > Click **Run workflow** > Select branch **master** > Type in a **Version** > Press **Run workflow**

Regards,
Patrick

**Comment by neroux** [comment]
@PatrickGotthard, that's great, but when will the fix be incorporated? ;)

It has been seven months since it was reported and half a year since a patch was provided.

**Comment by neroux** [comment]
> [Feel free to fix the warning and create a pull-requests :)](https://github.com/rometools/rome/issues/470#issuecomment-874821651)

@PatrickGotthard, I fully understand that the current maintainers might not have the time to continue supporting a project, but encouraging people to submit changes and then ignoring them for a year is neither very polite nor very productive either.

It might be time to look for project maintainers.

**Comment by PatrickGotthard** [comment]
Hi @neroux,

I've enhanced the OPML 2.0 parser now. The enhancement will be available in version 1.18.0.

Regards,
Patrick
